packages/@lwc/template-compiler/src/parser/index.ts

```diff
@@ -1012,7 +1025,7 @@ export default function parse(source: string, state: State): TemplateParseResult
     }
 
     function getRoot(element: IRElement): IRElement {
-        return element.parent ? getRoot(element.parent) : element;
+        return parentStack.length > 0 ? parentStack[0] : element;
```

Comment:
I would expect `getRoot` to be:

```suggestion
        return parentStack[0];
```

It is currently needed when `getRoot` is invoked on the root node itself. We can keep this for now, but it's something we should refactor after landing the new AST format.